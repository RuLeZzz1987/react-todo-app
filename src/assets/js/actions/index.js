export CategoryActions from './CategoryActions';
export TodoActions from './TodoActions';
export FilterActions from './FilterActions';