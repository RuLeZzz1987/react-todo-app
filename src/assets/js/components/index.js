import React, { Component } from 'react';
import Header from './header';
import Main from './main';
import { Category, Todo } from '../models';

class App extends Component {
    
    constructor(props) {
        super(props);
        
        this.state = {
            categories: [new Category({name: 'Category_1', isRoot: true, children: [new Todo({name: 'To-Do Item #1'}), new Category({name: 'Category_1_1'})]})]
        };
        
        this.updateCategories = (categories, cb) => this.setState({categories}, cb);
       
    }
    
    render() {
        return (
            <div>
                <Header
                    categories={this.state.categories}
                />
                <Main
                    categories={this.state.categories}
                    updateCategories={this.updateCategories}
                />
            </div>
        )
    }
}

export default App;