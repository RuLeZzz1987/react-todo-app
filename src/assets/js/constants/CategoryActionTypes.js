const Types = {
  ADD_CATEGORY: Symbol('ADD_CATEGORY'),
  EDIT_CATEGORY: Symbol('EDIT_CATEGORY'),
  REMOVE_CATEGORY: Symbol('REMOVE_CATEGORY'),
  TOGGLE_CATEGORY: Symbol('TOGGLE_CATEGORY'),
};

export default Types