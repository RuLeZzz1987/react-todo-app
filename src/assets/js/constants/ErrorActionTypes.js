const Types = {
  'SET_ERROR': Symbol('SET_ERROR'),
  'CLEAR_ERROR': Symbol('CLEAR_ERROR'),
  'SHOW_IN_POPUP': Symbol('SHOW_IN_POPUP'),
  'SET_ERROR_TYPE': Symbol('SET_ERROR_TYPE'),
};

export default Types