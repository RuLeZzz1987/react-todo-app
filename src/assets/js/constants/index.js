
export const CATEGORY = Symbol('category');

export const TODO = Symbol('todo');