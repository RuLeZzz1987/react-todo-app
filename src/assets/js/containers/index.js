export Header from './HeaderContainer';
export Categories from './CategoriesContainer';
export Category from './CategoryContainer';
export Editor from './EditorContainer';
export ModalError from './ModalErrorContainer';