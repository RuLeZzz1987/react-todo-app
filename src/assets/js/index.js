import React from 'react';
import { render } from 'react-dom';
import App from './components'
import '../styles/styles.scss';

render(<App/>, document.querySelector('#root'));