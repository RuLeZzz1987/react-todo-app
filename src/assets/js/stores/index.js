export TodoStore from './TodoStore';
export CategoryStore from './CategoryStore';
export FilterStore from './FilterStore';
export ErrorStore from './ErrorStore';