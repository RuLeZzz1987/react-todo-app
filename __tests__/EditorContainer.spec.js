/* eslint-disable prefer-arrow-callback */
import React from "react";
import renderer from "react-test-renderer";
import { Editor } from "../src/assets/js/containers";
import { ErrorStore } from "../src/assets/js/stores";
import { CATEGORY, TODO } from "../src/assets/js/constants";

describe("EditorContainer", function() {
  beforeEach(function() {
    this.width = 270;

    this.render = ({ type, add = jest.fn(), width }) =>
      renderer
        .create(
          <Editor
            type={type}
            placeholder={"Enter category title"}
            add={add}
            width={width}
          />
        )
        .toJSON();
  });

  it(
    "can render minimal editor container with mocked add function with type and width",
    function() {
      expect(this.render({ type: CATEGORY })).toMatchSnapshot();

      expect(this.render({ type: TODO, width: this.width })).toMatchSnapshot();
    }
  );

  it("can render editor container with error not matched type", function() {
    ErrorStore._state = {
      isError: true,
      message: "test message",
      showInPopup: false,
      type: CATEGORY
    };

    expect(this.render({ type: TODO, width: this.width })).toMatchSnapshot();
  });

  it("can render editor container with error shows in popup", function() {
    ErrorStore._state = {
      isError: true,
      message: "test message in popup",
      showInPopup: true,
      type: CATEGORY
    };

    expect(this.render({ type: CATEGORY })).toMatchSnapshot();
  });

  it("can render editor container with error", function() {
    ErrorStore._state = {
      isError: true,
      message: "test message in label",
      showInPopup: false,
      type: CATEGORY
    };

    expect(this.render({ type: CATEGORY })).toMatchSnapshot();
  });
});
