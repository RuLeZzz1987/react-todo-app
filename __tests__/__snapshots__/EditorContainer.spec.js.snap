exports[`EditorContainer can render editor container with error 1`] = `
<div
  className="editor">
  <input
    name="category"
    onChange={[Function]}
    placeholder="Enter category title"
    type="text"
    value="" />
  <label
    htmlFor="category">
    test message in label
  </label>
  <div
    className="add-button"
    onClick={[Function]}>
    <span
      className="add-label">
      ADD
    </span>
  </div>
</div>
`;

exports[`EditorContainer can render editor container with error not matched type 1`] = `
<div
  className="editor"
  style={
    Object {
      "width": "270px",
    }
  }>
  <input
    name="category"
    onChange={[Function]}
    placeholder="Enter category title"
    type="text"
    value="" />
  <div
    className="add-button"
    onClick={[Function]}>
    <span
      className="add-label">
      ADD
    </span>
  </div>
</div>
`;

exports[`EditorContainer can render editor container with error shows in popup 1`] = `
<div
  className="editor">
  <input
    name="category"
    onChange={[Function]}
    placeholder="Enter category title"
    type="text"
    value="" />
  <div
    className="add-button"
    onClick={[Function]}>
    <span
      className="add-label">
      ADD
    </span>
  </div>
</div>
`;

exports[`EditorContainer can render minimal editor container with mocked add function with type and width 1`] = `
<div
  className="editor">
  <input
    name="category"
    onChange={[Function]}
    placeholder="Enter category title"
    type="text"
    value="" />
  <div
    className="add-button"
    onClick={[Function]}>
    <span
      className="add-label">
      ADD
    </span>
  </div>
</div>
`;

exports[`EditorContainer can render minimal editor container with mocked add function with type and width 2`] = `
<div
  className="editor"
  style={
    Object {
      "width": "270px",
    }
  }>
  <input
    name="category"
    onChange={[Function]}
    placeholder="Enter category title"
    type="text"
    value="" />
  <div
    className="add-button"
    onClick={[Function]}>
    <span
      className="add-label">
      ADD
    </span>
  </div>
</div>
`;
